### PR TITLE
Clean up ready-to-run detection

### PR DESCRIPTION
**Pull request by sbomer**
- Use cecil's ILLibrary attribute (https://github.com/jbevain/cecil/pull/475)
- Factor common logic into a new function

@erozenfeld PTAL

### CODE REVIEW

File: linker/Linker/AssemblyUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using System;
+using Mono.Cecil;
+
+namespace Mono.Linker {
+
+	public class AssemblyUtilities {
```
Comment (marek-safar) [inline linker/Linker/AssemblyUtilities.cs]: Doesn't need to be public but should be static

File: linker/Linker/AssemblyUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+		public static bool IsReadyToRun (ModuleDefinition module)
```
Comment (marek-safar) [inline linker/Linker/AssemblyUtilities.cs]: Make this an extension method

Comment (sbomer) [reply linker/Linker/AssemblyUtilities.cs]: Done, thanks!

File: linker/Linker/AssemblyUtilities.cs (hunk 1 above)
Comment (sbomer) [reply linker/Linker/AssemblyUtilities.cs]: Made it static, but need to keep it public so I can use it from ILLink.Tasks.

**Comment by sbomer** [comment]
Eugene pointed out that SPC, while detected by this logic, isn't considered R2R. So I need to update the name of this check to something like IsCrossgend. Please hold off on merging this until I get a chance to do so.

**Comment by erozenfeld** [comment]
Yes, this logic detects all crossgen-ed assemblies not just R2R.

**Comment by sbomer** [comment]
I've renamed the check. I'll rename the tasks and msbuild items that depend on the check in https://github.com/mono/linker/pull/269, to keep these PRs somewhat independent.

**Comment by erozenfeld** [comment]
LGTM

**Pull request merged**